api/src/main/kotlin/no/nav/helsearbeidsgiver/inntektsmelding/api/hentforespoerselider/HentForespoerselIderRoute.kt

```diff
@@ -0,0 +1,89 @@
+package no.nav.helsearbeidsgiver.inntektsmelding.api.hentforespoerselider
+
+import io.ktor.http.HttpStatusCode
+import io.ktor.server.application.call
+import io.ktor.server.routing.Route
+import io.ktor.server.routing.post
+import io.prometheus.client.Summary
+import kotlinx.serialization.builtins.serializer
+import no.nav.helse.rapids_rivers.RapidsConnection
+import no.nav.helsearbeidsgiver.felles.domene.HentForespoerselIderResultat
+import no.nav.helsearbeidsgiver.felles.domene.ResultJson
+import no.nav.helsearbeidsgiver.felles.rapidsrivers.redis.RedisConnection
+import no.nav.helsearbeidsgiver.felles.rapidsrivers.redis.RedisPrefix
+import no.nav.helsearbeidsgiver.felles.rapidsrivers.redis.RedisStore
+import no.nav.helsearbeidsgiver.inntektsmelding.api.RedisPoller
+import no.nav.helsearbeidsgiver.inntektsmelding.api.RedisPollerTimeoutException
+import no.nav.helsearbeidsgiver.inntektsmelding.api.Routes
+import no.nav.helsearbeidsgiver.inntektsmelding.api.auth.ManglerAltinnRettigheterException
+import no.nav.helsearbeidsgiver.inntektsmelding.api.auth.Tilgangskontroll
+import no.nav.helsearbeidsgiver.inntektsmelding.api.logger
+import no.nav.helsearbeidsgiver.inntektsmelding.api.response.RedisTimeoutResponse
+import no.nav.helsearbeidsgiver.inntektsmelding.api.utils.receive
+import no.nav.helsearbeidsgiver.inntektsmelding.api.utils.respond
+import no.nav.helsearbeidsgiver.inntektsmelding.api.utils.respondBadRequest
+import no.nav.helsearbeidsgiver.inntektsmelding.api.utils.respondForbidden
+import no.nav.helsearbeidsgiver.inntektsmelding.api.utils.respondInternalServerError
+import no.nav.helsearbeidsgiver.utils.json.fromJson
+import java.util.UUID
+
+fun Route.hentForespoerselIderRoute(
+    rapid: RapidsConnection,
+    tilgangskontroll: Tilgangskontroll,
+    redisConnection: RedisConnection,
+) {
+    val hentForespoerselProducer = HentForespoerselIderProducer(rapid)
+    val redisPoller = RedisStore(redisConnection, RedisPrefix.HentForespoerselIder).let(::RedisPoller)
+
+    val requestLatency =
+        Summary
+            .build()
+            .name("simba_hent_forespoersel_ider_latency_seconds")
+            .help("hent forespoersel ider endpoint latency in seconds")
+            .register()
+
+    post(Routes.HENT_FORESPOERSEL_IDER) {
+        val transaksjonId = UUID.randomUUID()
+
+        val requestTimer = requestLatency.startTimer()
+
+        runCatching {
+            receive(HentForespoerselIderRequest.serializer())
+        }.onSuccess { request ->
+            logger.info("Henter forespørselIDer for vedtaksperiodeIDene: ${request.vedtaksperiodeIder}")
+            try {
+                tilgangskontroll.validerTilgangTilOrg(call.request, request.orgnr.verdi)
```

Comment:
Usikker på om det nye endepunktet bør ha noen form for tilgangskontroll (og i såfall hvilken type), så jeg slang inn denne typen her som sjekker om bruker har tilgang til orgnummeret (og så må den som kaller endepunktet passe på at man sender orgnummeret som hører til de vedtaksperiode-IDene man sender inn). Tanker?

Reply:
Dette er vel ikke en reell sjekk på at orgnr hører til vedtaksperiodeId da, antar jeg? Kan man egentlig bare sende inn et orgnr man har tilgang til, og så hvilken som helst ID for å bypasse? Sikkert litt kålete å få til noen bedre sjekk på vei inn. Kunne evt ha vasket resultatet etterpå, dersom man fikk hele forespørsler (med orgnr), ikke bare id - eller sendt orgnr videre, for å inkludere dette i kallet som henter forespørsler fra bro. Men det blir jo også litt styr.

Reply:
Jepp, godt innspill, det blir vasking på vei ut istedenfor 😄 🧹